### PR TITLE
feat(rules): Adding some rules

### PR DESCRIPTION
**Pull request by jeyrschabu**
- added a few asg rules
- added an age rule
- updated rules api

### CODE REVIEW

File: swabbie-aws/src/main/kotlin/com/netflix/spinnaker/swabbie/aws/autoscalinggroups/Rules.kt [hunk 1]
```diff
@@ -109,3 +111,122 @@ class ZeroInstanceInDiscoveryDisabledServerGroupRule(
...
+
+    val disabledTime = resource.disabledTime()!!
+    val moreThanDays = ruleDefinition?.parameters?.get("moreThanDays") as? Int
+    if (moreThanDays != null && disabledTime.isBefore(LocalDateTime.now(clock).minusDays(moreThanDays.toLong()))) {
+      return Result(
+        Summary(description = "Server Group ${resource.resourceId} has been out of balancer longer than $moreThanDays days.", ruleName = name())
+      )
+    }
+
+    return Result(
```
Comment (aravindmd) [inline swabbie-aws/src/main/kotlin/com/netflix/spinnaker/swabbie/aws/autoscalinggroups/Rules.kt]: not clear why is this needed ?

Comment (jeyrschabu) [reply swabbie-aws/src/main/kotlin/com/netflix/spinnaker/swabbie/aws/autoscalinggroups/Rules.kt]: `moreThanDays` is optional in this rule. This rule would enforce it if present otherwise it just asserts if the server group is disabled

Comment (jeyrschabu) [reply swabbie-aws/src/main/kotlin/com/netflix/spinnaker/swabbie/aws/autoscalinggroups/Rules.kt]: Will add that as a comment.

Comment (aravindmd) [reply swabbie-aws/src/main/kotlin/com/netflix/spinnaker/swabbie/aws/autoscalinggroups/Rules.kt]: gotcha !

File: swabbie-aws/src/test/java/com/netflix/spinnaker/swabbie/aws/autoscalinggroups/DisabledLoadBalancerRuleTest.kt [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      }
+
+    expectThat(rule.apply(asg, ruleDefinition).summary).isNull()
+
+    ruleDefinition = RuleDefinition()
+      .apply {
+        name = rule.name()
+        parameters = mapOf("moreThanDays" to 1)
+      }
+
```
Comment (aravindmd) [inline swabbie-aws/src/test/java/com/netflix/spinnaker/swabbie/aws/autoscalinggroups/DisabledLoadBalancerRuleTest.kt]: maybe worth adding the case for optional `moreThanDays` constraint here , which will validate just the serverGroup being disabled

Comment (jeyrschabu) [reply swabbie-aws/src/test/java/com/netflix/spinnaker/swabbie/aws/autoscalinggroups/DisabledLoadBalancerRuleTest.kt]: Reasonable. Will queue that up in another in-flight PR.
